File: public/examples/integrations/toolkits/mongodb/aggregate_documents_example_call_tool.js
import { Arcade } from "arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const TOOL_NAME = "MongoDB.AggregateDocuments";

const userId = "{arcade_user_id}";

const toolInput = {
  database_name: "my_database",
  collection_name: "users",
  pipeline: [
    JSON.stringify({"$match": {"status": "active"}}),
    JSON.stringify({"$group": {"_id": "$category", "count": {"$sum": 1}}}),
    JSON.stringify({"$sort": {"count": -1}})
  ],
  limit: 20
};

const response = await client.tools.execute({
  toolName: TOOL_NAME,
  input: toolInput,
  userId: userId,
});

console.log(response.output.value);

File: public/examples/integrations/mcp-servers/mongodb/aggregate_documents_example_call_tool.py
import json
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

TOOL_NAME = "MongoDB.AggregateDocuments"

user_id = "{arcade_user_id}"

tool_input = {
    "database_name": "my_database",
    "collection_name": "users",
    "pipeline": [
        json.dumps({"$match": {"status": "active"}}),
        json.dumps({"$group": {"_id": "$category", "count": {"$sum": 1}}}),
        json.dumps({"$sort": {"count": -1}})
    ],
    "limit": 20
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=user_id,
)

print(response.output.value)
